blackbird_python/blackbird/listener.py

```diff
@@ -99,6 +99,7 @@ def __init__(self, expr):
         * :attr:`func_str`
         """
         regref_symbols = list(expr.free_symbols)
+        self.expr = expr
```

Comment:
Worth adding an attribute docstring here as for the others (even if short), and adding this to the list of attributes (starting on line 97).